### ISSUE TITLE
Auto-complete tries to complete absolutely every word

### ISSUE THREAD

**Issue by krishunt** [emmetio/nova-plugin#32]
When auto-complete is turned on, Emmet tries to complete every single word I type, and it's very annoying because the popups never go away.

<img width="474" alt="Emmet Auto Complete" src="https://user-images.githubusercontent.com/5502749/107714640-1dff1b80-6c7a-11eb-9892-0fa63ec1c274.png">

The only workaround I've found is to disable auto-complete entirely, and then manually auto-complete with a keyboard shortcut. But this is obviously an undesirable solution. Can auto-complete be set to only suggest auto-completing recognized keywords?

**Comment by sergeche** [comment]
Yes, will add this option. On the other side, in JSX context you would expect the same behaviour as above since it’s a perfectly valid React component name

**Comment by davidbdeath** [comment]
I also have this complaint. I had to disable emmet to write a SASS file. Also, consider not having enter autocomplete. It's natural to press enter to go to the next line. It is highly inefficient to require pressing ESC before i can hit enter.

**Comment by arghyaBiswasDev** [comment]
Same here. Wish this could be fixed.

**Comment by hrvstr** [comment]
Any updates on this issue? It's been almost two years!

**Comment by davidbdyer** [comment]
Doesn't look like it. The only edit I see in the code is to add support for another language. Looks like they don't care about Nova. To be honest Panic shouldn't be leaving some of these make or break plugs up to third parties. Their JS support sadly  is sub par to VS Codes.

**Comment by hrvstr** [comment]
Exactly, I send a mail to Panic yesterday asking them about this issue. Having Nova without proper Emmet support is kind of pointless IMO.

**Comment by sergeche** [comment]
Unfortunately, Nova has a very limited API, compared to VSCode or Sublime Text. When I developed this extension, it was really hard to implement completions with preview as in other editors. And I sent my suggestions to Panic, don’t know if they were implemented already (my Nova license has expired, can’t try updated API).

The official solution for JS/JSX for Emmet is to use `<` prefix for Emmet completions. Here’s how it works in Sublime Text:

https://user-images.githubusercontent.com/93595/211773252-81cfacea-5712-4540-a3a3-6cd05cd132f4.mov

In Nova, `<` symbol can’t trigger suggestions, as well as any following characters won’t trigger it. So there’s not much I can do with it to provide better UX for JSX

**Comment by hrvstr** [comment]
Thank you for your reply and for creating this extension in the first place.

You can request a free license (and renew it every year) as an extension developer. See this blog post for all the details: https://panic.com/blog/free-copy-of-nova-for-extension-developers/

I am not sure about the API but I think I've seen a couple of enhancements in the release notes here: https://nova.app/releases/

I would really love to see some improvements here because Nova has the potential to be an awesome editor but I can't use it without Emmet. I will post here once I hopefully get a reply from Panic.

https://user-images.githubusercontent.com/4610008/212024506-0aba6202-3be4-4a38-bf47-2105ee51f528.mov

**Comment by rupl** [comment]
Could the mass-autocomplete possibly be disabled until a `-` character is entered?

For example if I type `faketag` I definitely don't expect that to autocomplete to an element. However once a `-` character is encountered, this is officially a [custom element](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name) since default HTML tags don't use the dash. I think catering to JSX is detrimental. As others have said, basically makes the plugin unusable for the people using web standards.

```html
<!-- no expansion for random words -->
faketag

<!-- expand known elements in HTML spec -->
p <p></p>
ul <ul></ul>
li <li></li>

<!-- also expand once dash is typed -->
my-element <my-element></my-element>
```

**Comment by sergeche** [comment]
I’ll try to figure out if it’s possible to filter completions somehow. At the time when extention was written, Nova API was very limited. Maybe there are new methods to available
